core/src/vaults.ts

```diff
@@ -0,0 +1,378 @@
+    vault: Vault
+): Promise<VaultTransactionNotLiquidated> => {
+    const debtDai = vault.initialDebtDai.multipliedBy(vault.stabilityFeeRate);
+    const { liquidatiorContractAddress } = await fetchCollateralLiquidationLimitsAndLiquidatorAddress(
+        network,
+        vault.collateralType
+    );
+    const { incentiveCombinedDai, incentiveConstantDai, incentiveRelativeDai } = await fetchVaultLiquidationIncentive(
+        network,
+        liquidatiorContractAddress,
+        debtDai
+    );
+
+    const { liquidationRatio, oracleAddress } = await fetchLiquidationRatio(network, vault.collateralType);
+    const collateralizationRatio = vault.collateralAmount
+        .multipliedBy(vault.minUnitPrice)
+        .multipliedBy(liquidationRatio)
+        .dividedBy(debtDai)
+        .toNumber();
+    const proximityToLiquidation = collateralizationRatio - liquidationRatio;
+    const { transactionFeeLiquidationEth, transactionFeeLiquidationDai } = await getApproximateLiquidationFees(
+        network
+    );
+    const osmPrices = await getOsmPrices(network, oracleAddress);
+    const { nextUnitPrice, nextPriceChange, currentUnitPrice } = osmPrices
+        ? osmPrices
+        : {
+              nextUnitPrice: new BigNumber(NaN),
+              currentUnitPrice: new BigNumber(NaN),
+              nextPriceChange: undefined,
+          };
+    const state = proximityToLiquidation < 0 ? 'liquidatable' : 'not-liquidatable';
+    return {
+        ...vault,
+        liquidationRatio,
+        collateralizationRatio,
+        proximityToLiquidation,
+        incentiveConstantDai,
+        incentiveCombinedDai,
+        incentiveRelativeDai,
+        grossProfitDai: incentiveCombinedDai,
+        netProfitDai: incentiveCombinedDai.minus(transactionFeeLiquidationDai),
+        transactionFeeLiquidationDai,
+        debtDai,
+        nextUnitPrice,
+        nextPriceChange,
+        currentUnitPrice,
+        transactionFeeLiquidationEth,
+        state,
+    };
+};
+
+export const enrichVaultWithTransactonInformation = memoizee(_enrichVaultWithTransactonInformation, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+const _getVaultTransaction = async (network: string, vault: Vault): Promise<VaultTransaction> => {
+    const { isLiquidated, liquidationDate, transactionHash, auctionId } = await isVaultLiquidated(network, vault);
+    if (isLiquidated) {
+        return {
+            ...vault,
+            state: 'liquidated',
+            liquidationDate: liquidationDate as Date,
+            transactionHash: transactionHash as string,
+            auctionId: auctionId as string,
+        };
+    }
+    return await enrichVaultWithTransactonInformation(network, vault);
+};
+
+export const getVaultTransaction = memoizee(_getVaultTransaction, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+export const liquidateVault = async (network: string, vault: Vault, incentiveBeneficiaryAddress: string) => {
```

Comment:
Can we instead accept `collateralType` and `vaultAddress` here? `incentiveBeneficiaryAddress` should be optional and is taken form the signer if not provided

Reply:
- [ ] Can we instead accept `collateralType` and `vaultAddress` here?

Reply:
reading is hard 😞